File: app/src/main/java/org/deiverbum/app/data/entity/relation/MisaWithLecturas.java
package org.deiverbum.app.data.entity.relation;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.data.entity.BibleReadingEntity;
import org.deiverbum.app.data.entity.MassReadingEntity;
import org.deiverbum.app.model.MassReading;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
public class MisaWithLecturas {
    @Embedded
    public MassReadingEntity misaLectura;

    @Relation(
            parentColumn = "readingFK",
            entityColumn = "readingID",
            entity = BibleReadingEntity.class
    )
    public MassReadingWithBook lectura;

    public MassReading getDomainModel() {
        MassReading theModel = lectura.getDomainModelMisa();
        theModel.setTema(misaLectura.getTema());
        theModel.setOrden(misaLectura.getOrden());
        return theModel;
    }

}

File: app/src/main/java/org/deiverbum/app/data/entity/relation/MisaWithLecturas.kt
package org.deiverbum.app.data.entity.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.data.entity.BibleReadingEntity
import org.deiverbum.app.data.entity.MassReadingEntity
import org.deiverbum.app.model.MassReading

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
class MisaWithLecturas {
    @JvmField
    @Embedded
    var misaLectura: MassReadingEntity? = null

    @JvmField
    @Relation(
        parentColumn = "readingFK",
        entityColumn = "readingID",
        entity = BibleReadingEntity::class
    )
    var lectura: MassReadingWithBook? = null
    val domainModel: MassReading?
        get() {
            val theModel = lectura?.domainModelMisa
            theModel!!.tema = misaLectura!!.tema
            theModel!!.setOrden(misaLectura!!.orden)
            return theModel
        }
}